tests/unittests/test_date_out_of_range.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+from unittest import mock
+from tap_jira.streams import Stream
+
+class TestOutOfRangeDate(unittest.TestCase):
+    @mock.patch("tap_jira.streams.singer.utils.now", return_value="2022-05-23T09:16:11.356670Z")
+    @mock.patch("tap_jira.streams.singer.write_record")
+    @mock.patch("tap_jira.streams.Context")
+    @mock.patch("tap_jira.streams.metadata")
+    def test_out_of_range_date(self, mock_metadata, mock_Context, mock_write_record, mock_now):
+        """
+        Verify that the tap successfully skips record for out of range date value.
+        """
+        mock_records =[
+            {"updated": "2017000-09-05T19:51:03.159Z", "id": "1","id2": "11"},  # year out of range
+            {"updated": "2001-09-05T19:51:03.159000", "id": "2","id2": "12"},  # Correct record
+            {"updated": "2001-13-05T19:51:03.159000", "id": "3","id2": "13"},  # month out of range
+            {"updated": "2007-09-45T19:51:03.159000z", "id": "4","id2": "14"},  # day out of range
+            {"updated": "2006-01-29T01:62:01.99999", "id": "5","id2": "16"},  # minutes out of range
+            {"updated": "2011-09-05T19:51:03.159000", "id": "6","id2": "15"},  # Correct record
+            {"updated": "2009-09-10T26:51:03.159000Z", "id": "7","id2": "17"},  # hour out of range
+        ]
```

Comment:
I think last record should be valid one to verify that all records skipped with invalid format and all valid ones replicated.

Reply:
Yes, thats right, hence we have just 2 call counts [here](https://github.com/singer-io/tap-jira/pull/87/files#diff-38c5ac42c3c3fcae142935dae6ed190e2647fea82e6808ebc0ced96055a3387bR39)

Reply:
Updated the unittest as per your suggestions